pkg/compact/compact.go

```diff
@@ -1201,7 +1233,19 @@ func (c *BucketCompactor) Compact(ctx context.Context) (rerr error) {
 			go func() {
 				defer wg.Done()
 				for g := range groupChan {
-					shouldRerunGroup, _, err := g.Compact(workCtx, c.compactDir, c.planner, c.comp)
+					workCtx = tracing.CopyTraceContext(workCtx, ctx)
+					// adding a parent span to ctx for overall compaction
+					// do not need to add a separate span since the parent span is not used separately here - can parent span be passed from original compaction function?
+					var groupSpan tracing.Span
+					groupSpan, ctx = tracing.StartSpan(workCtx, "compaction", opentracing.Tags{"group key": g.Key()})
+					defer groupSpan.Finish() // added to avoid parent spans being finished before child
```

Comment:
I guess each group span is used to check each compaction group. So you cannot use defer here as it is a for loop. Let's finish it explicitly.

Reply:
Btw I don't get why we need this span TBH. We already have a span for compaction at L1245. Then what's this group span for? Isn't that the same thing?

Reply:
I had planned on a separate span for each group and hence, had explicitly created it. But, I see your point and I've removed it.
Thanks!